datafusion/expr/src/udaf.rs

```diff
@@ -225,10 +225,10 @@ impl AggregateUDF {
 
     /// See [`AggregateUDFImpl::with_beneficial_ordering`] for more details.
     pub fn with_beneficial_ordering(
-        self,
+        self: Arc<Self>,
```

Comment:
What's the benefit of using Arc here?

```suggestion
        &self
```